app/assets/javascripts/discourse/app/static/prosemirror/extensions/mention.js

```diff
@@ -0,0 +1,76 @@
+import { mentionRegex } from "pretty-text/mentions";
+import { isBoundary } from "discourse/static/prosemirror/lib/markdown-it";
+
+/** @type {RichEditorExtension} */
+const extension = {
+  nodeSpec: {
+    mention: {
+      attrs: { name: {} },
+      inline: true,
+      group: "inline",
+      content: "text*",
+      atom: true,
+      draggable: true,
+      selectable: false,
+      parseDOM: [
+        {
+          tag: "a.mention",
+          preserveWhitespace: "full",
+          getAttrs: (dom) => {
+            return { name: dom.getAttribute("data-name") };
+          },
+        },
+      ],
+      toDOM: (node) => {
+        return [
+          "a",
+          { class: "mention", "data-name": node.attrs.name },
+          `@${node.attrs.name}`,
+        ];
+      },
+    },
+  },
+
+  inputRules: [
+    {
+      // TODO(renato): pass unicodeUsernames?
+      match: new RegExp(`(?<=^|\\W)(${mentionRegex().source}) $`),
```

Comment:
The positive lookbehind is not supported in older Safari browsers (< 16.4) and can crash the application (see ee3ccc5)

We currently support Safari 15.7+

Reply:
Done, thanks!